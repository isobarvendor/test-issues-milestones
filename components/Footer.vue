<template>
  <footer>
      <div class="container">
        <p v-html="footerText"></p>
        <p>
          <a href="">Privacy Policy</a>
          <a href="">Terms and conditions</a>
        </p>
      </div>
    </footer>
</template>

<script>
export default {
    name:"Footer",
    props:{
      data:null
    },
    data(){
      return {
        footerText:''
      }
    },
    mounted(){
      var text = this.data.footerText;
      this.footerText = text.replace(/The Coca-Cola Company/g, 'The&nbsp;Coca&#8209;Cola&nbsp;Company');
    }
}
</script>

<style>

</style>