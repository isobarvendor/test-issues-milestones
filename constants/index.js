let language = {
  en: {
    browserTitle: "Turn Up Your Rhythm",
    meta: [
      {
        hid: "Turn Up Your Rhythm",
        name: "Turn Up Your Rhythm",
        content: "Home page Coke - Turn Up Your Rhythm"
      }
    ],
    periodDescription: "",
    signInButton: "SIGN IN <BR/><small>to win prizes</small>",
    menu: {
      home: "Home",
      privacy: "Privacy",
      terms: "Terms",
      logout: "Logout",
      myRewards: "My Rewards"
    },
    countdown: {
      days: "DAYS",
      hrs: "HRS",
      mins: "MINS",
      secs: "SECS"
    },
    submissionText: {
      header: "Enter your code",
      hello: "Hello",
      siginHeader:
        "Sign in or register a Coca-Cola <br>account to enter the draw",
      name: "Name",
      email: "Email",
      phoneNumber: "Phone Number",
      acceptTerm:
        'I accept the <a href="/tnc">Terms and Conditions</a> of this this redemption.',
      acceptPrivacy:
        'I accept the <a href="/privacy">Privacy Policy</a> of this redemption.',
      declareAge: "I declare that I am above 13  years old.",
      enterCode: "Enter Unique Code",
      tooltipText:
        "Enter the code found under the cap/tab of your Coca Cola purchase here",
      phoneTooltip:
        "Please input the correct phone number so that Coca-Cola can contact you directly incase you are the winner",
      buttonText: "Collect your prize",
      errorTerm: "Please accept our terms and conditions",
      errorPolicy: "Please accept our privacy policies",
      errorDeclare: "Please declare that you are above 13  years old",
      errorPinCode: "Oops your pin code is invalid or already redeemed",
      errorPinCode1: "Oops your pin code is invalid or already redeemed",
      errorPinCode2: "Oops your pin code is invalid or already redeemed",
      errorPinCode3: "Oops your pin code is invalid or already redeemed",
      errorPinCode4: "Oops your pin code is invalid or already redeemed",
      errorZaloPrize: "Redemption Limit for zalo prize has been reached",
      errorNormalPrize: "Redemption Limit for normal prize has been reached",
      errorAPI: "Oops something went wrong please try again",
      errorMaxPhone: "You reach maximum phone number length",
      errorRequiredPhone: "The phone number is required",
      participateAgain: "Participate again",
      listenNow: "Listen now",
      redeemPrize: "Redeem Prize",
      continue: "Continue with",
      note:
        "Please call 1800 555 584 (VND 1500/min, from 9:00 to 17:00, weekdays) to reveive redemption instructions",
      startRedeeming: "Start Redeeming",
      redeemNextPrize: "Redeem next prize",
      textPage:
        "You have entered the Entry Code from Coca-Cola 300ml PET bottle or Coca-Cola 235ml can, you will have 2 lucky draw opportunities",
      hardLuckHeader: "",
      hardLuckTitle:
        "Bạn chưa may mắn lần này. Nhưng vẫn còn nhiều quà hấp dẫn đang chờ. Hãy tiếp tục khui Coke và tham gia bạn nhé!",
      hardLuckImage: "/img/landing/hardluck.png"
    },
    winnerText: {
      header: "WINNER LIST",
      nowinner: "Coming soon"
    },
    globalError: {
      errorAPI: "Oops something went wrong",
      tryagain: "Please try again"
    },
    footer: {
      privacy: "Privacy Policy",
      terms: "Terms and conditions"
    }
  },
  kh: {
    browserTitle: "បញ្ចេញភាពរស់រវើករបស់អ្នក",
    meta: [
      {
        hid: "បញ្ចេញភាពរស់រវើករបស់អ្នក",
        name: "បញ្ចេញភាពរស់រវើករបស់អ្នក",
        content: "Home page Coke - បញ្ចេញភាពរស់រវើករបស់អ្នក"
      }
    ],
    periodDescription:
      "ចាប់ពីថ្ងៃទី 04 កញ្ញា 2021 ដល់ថ្ងៃទី 30 តុលា 2021 គ្រប់ផលិតផលកូកា-កូឡា អ្នកអាចមានឱកាសស្តាប់ចម្រៀងថ្មីៗ ពី Pleng (Unbox Music Playlist) និងរង្វាន់ជាច្រើនទៀត ដោយគ្រាន់តែអនុវត្តតាមជំហានដូចខាងក្រោម",
    signInButton: "ចុះឈ្មោះ<BR/><small>ដើម្បីមានឱកាសឈ្នះ</small>",
    menu: {
      home: "ទំព័រដើម",
      privacy: "គោលការណ៍ឯកជន",
      terms: "លក្ខន្ដិកនៃការចូលរួម",
      logout: "Logout",
      myRewards: "My Rewards"
    },
    countdown: {
      days: "DAYS",
      hrs: "HRS",
      mins: "MINS",
      secs: "SECS"
    },
    submissionText: {
      header: "វាយបញ្ចូលលេខកូដ",
      hello: "Hello",
      siginHeader: " ចុះឈ្មោះដើម្បីចូលរួម",
      name: "ឈ្មោះ",
      email: "អុីម៉េល",
      phoneNumber: "លេខទូរស័ព្ទ",
      acceptTerm: "ខ្ញុំយល់ព្រមទទួលយកនូវលក្ខន្ដិកនេះ",
      acceptPrivacy: "ខ្ញុំយល់ព្រមទទួលយកនូវគោលការណ៍ឯកជនភាពនេះ",
      declareAge: "ខ្ញុំសូមប្រកាសថាខ្ញុំមានអាយុលើស 13 ឆ្នាំ",
      enterCode: "វាយបញ្ចូលលេខកូដ",
      tooltipText:
        "វាយបញ្ចូលលេខកូដដែលឃើញពីក្រោមគម្របដប ឬក្រវិលកំប៉ុងកូកា-កូឡារបស់អ្នកនៅទីនេះ",
      phoneTooltip:
        "សូមដាក់លេខទូរស័ព្ទពិតប្រាកដរបស់អ្នកដើម្បីក្រុមការងារកូកា-កូឡាអាចទាក់ទងទៅលោកអ្នកនៅពេលដែលឈ្នះបានទូរស័ព្ទ និង​កាសទូរស័ព្ទ",
      buttonText: "ទទួលយករង្វាន់របស់អ្នក",
      errorTerm: "សូមចុចយល់ព្រមទទួលយកនូវលក្ខន្ដិកៈនៃការចូលរួមកម្មវិធី",
      errorPolicy: "សូមចុចយល់ព្រមទទួលគោលការណ៍ឯកជនភាព",
      errorDeclare: "សូមប្រកាសថាអ្នកមានអាយុលើពី ១៣ ឆ្នាំ",
      errorPinCode: " លេខកូដមិនត្រឹមត្រូវ ឬបានប្តូររង្វាន់រួចហើយ",
      errorPinCode1: " លេខកូដមិនត្រឹមត្រូវ ឬបានប្តូររង្វាន់រួចហើយ",
      errorPinCode2: " លេខកូដមិនត្រឹមត្រូវ ឬបានប្តូររង្វាន់រួចហើយ",
      errorPinCode3: " លេខកូដមិនត្រឹមត្រូវ ឬបានប្តូររង្វាន់រួចហើយ",
      errorPinCode4: " លេខកូដមិនត្រឹមត្រូវ ឬបានប្តូររង្វាន់រួចហើយ",
      errorZaloPrize: "Redemption Limit for zalo prize has been reached",
      errorNormalPrize: "Redemption Limit for normal prize has been reached",
      errorAPI: " បញ្ហាបច្ចេទេស សូមព្យាយាមម្តងទៀត",
      errorMaxPhone: "អ្នកបញ្ជូលលេខទូរស័ព្ទគ្រប់ខ្ទង់",
      errorRequiredPhone: " តម្រូវឲ្យមានលេខទូរស័ព្ទ",
      participateAgain: " ចូលរួមម្តងទៀត",
      listenNow: "ស្តាប់ឥឡូវនេះ",
      redeemPrize: " ប្តូរយករង្វាន់",
      continue: " ចុចបន្ត",
      note: "សូមទាក់ទងលេខ 1-800-20-2222 ដើម្បីដឹងពីរបៀបប្តូរយករង្វាន់",
      startRedeeming: "ចាប់ផ្តើមប្តូរយករង្វាន់",
      redeemNextPrize: "ប្តូរយករង្វាន់បន្ទាប់",
      thankyouMessage: "<h2>អបអរសាទរ! អ្នកឈ្នះរង្វាន់ទ្វេរដង!<h2/>",
      textPage:
        "អ្នកបានបញ្ចូលលេខកូដពីដបកូកា-កូឡាដបជ័រ 350ml អ្នកនឹងមានឱកាសចាប់រង្វាន់ 2 ដង",
      hardLuckHeader: "មានបញ្ហាបច្ចេទេស",
      hardLuckTitle: "",
      hardLuckImage: "",
      theCodeFieldIsRequired: "សូមបំពេញលេខកូដទីនេះ"
    },
    winnerText: {
      header: "តារាងឈ្មោះអ្នកឈ្នះ",
      nowinner: "មកដល់ឆាប់ៗ នេះ"
    },
    globalError: {
      errorAPI: "មានបញ្ហាបច្ចេទេស",
      tryagain: " សូមព្យាយាមម្តងទៀត"
    },
    footer: {
      privacy: "គោលការណ៍ឯកជន",
      terms: "លក្ខន្ដិកនៃការចូលរួម"
    }
  }
  // vn:{
  //     browserTitle:"Turn Up Your Rhythm",
  //     meta: [
  //       {
  //         hid: "Turn Up Your Rhythm",
  //         name: "Turn Up Your Rhythm",
  //         content: "Home page Coke - Turn Up Your Rhythm"
  //       }
  //     ],
  //     periodDescription:"Rất tiếc thời gian CTKM Bừng Giai Điệu Hứng Khởi đã kết thúc vào ngày 06/06/2021",
  //     signInButton:"ĐĂNG NHẬP <BR/><small>để trúng thưởng</small>",
  //     menu:{
  //       home:"Trang Chủ",
  //       privacy:"Điều khoản bảo mật",
  //       terms:"Điều kiện & Điều khoản áp dụng",
  //       logout:"đăng xuất",
  //       myRewards:"Giải thưởng của bạn"
  //     },
  //     countdown:{
  //       days:"DAYS",
  //       hrs:"HRS",
  //       mins:"MINS",
  //       secs:"SECS",
  //     },
  //     submissionText:{
  //       header:"",
  //       hello:"Hello",
  //       siginHeader:'Đăng kí và bắt đầu tham gia chương trình rút thăm trúng thưởng',
  //       name:"Họ & Tên",
  //       email:"Email",
  //       phoneNumber:"Số điện thoại",
  //       acceptTerm:'Tôi đồng ý với <a href="/tnc" target="_blank">Điều kiện và Điều khoản được áp dụng</a> cho chương trình khuyến mãi',
  //       acceptPrivacy:'Tôi đồng ý với <a href="/privacy" target="_blank">Điều khoản Bảo mật</a> của chương trình khuyến mãi',
  //       declareAge:'Tôi cam đoan đã đủ hoặc trên 18 tuổi',
  //       enterCode:'Nhập mã ngay',
  //       tooltipText:'Nhập mã dự thưởng bên dưới nắp khoen hay nắp chai của sản phẩm Coca-Cola có khuyến mãi',
  //       phoneTooltip:"Vui lòng nhập chính xác số điện thoại để BTC có thể liên lạc trong trường hợp bạn là người trúng giải",
  //       buttonText:'Nhận quà thưởng',
  //       errorTerm:'Vui lòng chấp nhận các điều khoản và điều kiện của chúng tôi',
  //       errorPolicy:"Vui lòng chấp nhận chính sách bảo mật của chúng tôi",
  //       errorDeclare:"Vui lòng tuyên bố rằng bạn trên 18 tuổi",
  //       errorPinCode:"Xin lỗi Mã Tham Dự này không đúng. Vui lòng kiểm tra lại.",
  //       errorPinCode1:"Xin lỗi Mã Tham Dự này không đúng. Vui lòng kiểm tra lại.",
  //       errorPinCode2:"Xin lỗi Mã Tham Dự này đã được sử dụng. Vui lòng kiểm tra lại.",
  //       errorPinCode3:"Xin lỗi Mã Tham Dự này không đúng. Vui lòng kiểm tra lại.",
  //       errorPinCode4:"Xin lỗi Mã Tham Dự này không đúng. Vui lòng kiểm tra lại.",
  //       errorZaloPrize:"Redemption Limit for zalo prize has been reached",
  //       errorNormalPrize:"Redemption Limit for normal prize has been reached",
  //       errorAPI:"Rất tiếc. Đã xảy ra lỗi. Vui lòng thử lại",
  //       errorMaxPhone:"Bạn đạt đến độ dài số điện thoại tối đa",
  //       participateAgain:"Participate again",
  //       listenNow:"Nghe nhạc ngay",
  //       redeemPrize:"ĐỔI GIẢI THƯỞNG TẠI ĐÂY",
  //       continue:"Tiếp tục với",
  //       note:"Vui lòng gọi hotline 1900555584 (1500VND/ phút từ 9:00 đến 17:00, ngày hành chính) để nhận hướng dẫn thủ tục nhận giải thưởng.",
  //       startRedeeming:"Bắt đầu rút thăm",
  //       redeemNextPrize:"Tiếp tục đổi giải",
  //       errorRequiredPhone:"The phone number is required",
  //       textPage:"Bạn vừa nhập Mã dự thưởng từ sản phẩm chai nhựa Coca-Cola 300ml hoặc Lon Coca-Cola 235ml. Bạn có 2 cơ hội rút thăm trúng thưởng",
  //       hardLuckHeader:"",
  //       hardLuckTitle:"Lượt nghe một bài nhạc độc quyền của Coca-Cola",
  //       hardLuckImage:"https://d1tnwdlrciourl.cloudfront.net/prizeImages/1e0b8kkwap0rj/kkwbxo87/ExclusiveSong.png",
  //       hardLuckMusic:"https://elasticbeanstalk-prod-appcontent-016176.s3-ap-southeast-1.amazonaws.com/can_mong_tay_master_11-4_fix.mp3",
  //     },
  //     winnerText:{
  //       header:"WINNER LIST",
  //       nowinner:"Coming soon"
  //     },
  //     globalError:{
  //       errorAPI:"Oops something went wrong",
  //       tryagain:"Please try again"
  //     },
  //     footer:{
  //       privacy:"Điều khoản bảo mật",
  //       terms:"Điều kiện & Điều khoản áp dụng"
  //     }

  //   }
};

export const translation = language[process.env.language];
