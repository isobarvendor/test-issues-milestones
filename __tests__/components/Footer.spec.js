
import {shallowMount  } from '@vue/test-utils'

import Footer from "@/components/Footer";
import {translation} from "@/constants/index"

const factory = () => {
  return shallowMount(Footer, {
    data() {
      return {
        footer:translation.footer
      };
    },
    propsData: {
      data :{
       copyright:{
         copyright:"tes"
       }
      }
    }
  });
};

describe("Footer", () => {


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
