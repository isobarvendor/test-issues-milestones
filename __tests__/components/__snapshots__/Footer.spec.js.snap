// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders properly 1`] = `
<footer>
  <div class="container">
    <p class="footer-text">tes</p>
    <p><a href="/privacy">Privacy Policy</a> <a href="/tnc">Terms and conditions</a></p>
  </div>
</footer>
`;
